hazelcast/src/main/java/com/hazelcast/map/impl/BasicMapContextQuerySupport.java

```diff
@@ -142,9 +142,10 @@ public Set queryLocalMember(String mapName, Predicate predicate, IterationType i
             if (partitionIds.isEmpty()) {
                 return result;
             }
-        } catch (QueryResultSizeExceededException e) {
-            throw ExceptionUtil.rethrow(e);
         } catch (Throwable t) {
+            if (t.getCause() instanceof QueryResultSizeExceededException) {
```

Comment:
The whole code is about preventing the second block (retry on all partitions) if, and only if, the `QueryResultSizeExceededException` was thrown in the first block (query on members).

The problem is, that we don't want to catch and rethrow all exceptions, just my custom exception. I don't know this by heart, but there are situations where the first part will fail (I think some slow `MapLoader` scenarios). So we cannot just catch all exceptions and stop. So we won't get rid of catching my custom exception in all these parts, even if we convert it to another exception type.

Or did I get your idea wrong?